### DISCUSSION TITLE
NNUE file compatibility?

### DISCUSSION THREAD

**Discussion by nguyenpham** [official-stockfish/Stockfish#4832] [Q&A] [answered]
I have read somewhere that SF has already so many architectures for NNUE. Just wondering about the compatibility of NNUE files: can new versions of SF read all older NNUE files and vice versa?

**Comment by PGG106** [comment, marked as answer]
No, different architectures require different inference code so you shouldn't expect neither backwards nor forwards compatibility.

**Reply by nguyenpham** [reply]
What happens if I load an incomparable NNUE file for SF? Will it crash or say something?

**Reply by PGG106** [reply]
i can't say for certain since i haven't tried, it will do for sure one of the 3: complain about the net and refuse to work, crash or return bogus evals because it's reading random garbage

**Reply by nguyenpham** [reply]
Is there any meta data/information on NNUE files, say, headers, signatures, flags... that we could use to check create time, versions, architectures, compatibilities...?

**Comment by Sopel97** [comment]
It will crash on first use and there's no way to make it more graceful because UCI lacks error handling.
